.github/workflows/publish.yml

```
@@ -0,0 +1,22 @@
+name: Publish to NPM
+on:
+  push:
+    branches:
+      - main
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Set up Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18.x'
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies and build
+        run: npm install && npm run build
+      - name: Publish package on NPM 📦
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

```suggestion
          NODE_AUTH_TOKEN: ${{ secrets.WAVESHQ_NPM_TOKEN }}
```
reference: https://github.com/WavesHQ/walletkit/blob/main/.github/workflows/release.yml